src/TLuaInterpreterMudletObjects.cpp

```diff
@@ -2555,3 +2556,114 @@ int TLuaInterpreter::tempTrigger(lua_State* L)
     return 1;
 }
 
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#getProfiles
+int TLuaInterpreter::getProfiles(lua_State* L)
+{
+    auto& hostManager = mudlet::self()->getHostManager();
+    const QStringList profiles = QDir(mudlet::getMudletPath(mudlet::profilesPath))
+                                   .entryList(QDir::Dirs | QDir::NoDotAndDotDot, QDir::Name);
+
+    lua_newtable(L);
+
+    for (const auto& profile : profiles) {
+        lua_pushstring(L, profile.toUtf8().constData());
+        lua_newtable(L);
+
+        QString url = mudlet::self()->readProfileData(profile, qsl("url"));
+        QString port = mudlet::self()->readProfileData(profile, qsl("port"));
+
+        // if url/port haven't been written to disk yet (which is what happens
+        // when a default profile is opened for the first time), fetch this data from game details
+        if (url.isEmpty()) {
+            auto it = TGameDetails::findGame(profile);
+            if (it != TGameDetails::scmDefaultGames.end()) {
+                url = (*it).hostUrl;
+            }
+        }
+        if (port.isEmpty()) {
+            auto it = TGameDetails::findGame(profile);
+            if (it != TGameDetails::scmDefaultGames.end()) {
+                port = QString::number((*it).port);
+            }
+        }
+
+        if (!url.isEmpty()) {
+            lua_pushstring(L, "host");
+            lua_pushstring(L, url.toUtf8().constData());
+            lua_settable(L, -3);
+        }
+
+        if (!port.isEmpty()) {
+            lua_pushstring(L, "port");
+            lua_pushstring(L, port.toUtf8().constData());
+            lua_settable(L, -3);
+        }
+
+        auto host = hostManager.getHost(profile);
+        const bool loaded = host != nullptr;
+        lua_pushstring(L, "loaded");
+        lua_pushboolean(L, loaded);
+        lua_settable(L, -3);
+
+        if (loaded) {
+            auto [hostName, hostPort, connected] = host->mTelnet.getConnectionInfo();
+
+            lua_pushstring(L, "connected");
+            lua_pushboolean(L, connected);
+            lua_settable(L, -3);
+        }
```

Comment:
Wouldn't it be cleaner for any end-user code to always include the `connected` key but always set it to `false` if `loaded` is not true - alternatively - we do not have to use `booleans` we could instead code them together as `state` with - say values of: `inactive`/`loaded`/`connected`?

Reply:
Could be done either way, let's wait on player feedback for which version is preferred.

Reply:
I'm 50/50.

if connected then

vs

if connected=true


I'd defer to you devs as to what the mudlet/lua standard is around this really, either works here.

Didn't test yet, had a power outage at home all day. Will test Thurs or Fri though for sure.

Reply:
> Could be done either way, let's wait on player feedback for which version is preferred.

Is that reasonable though, as it would break the backwards compatibility if we want and changed it after the function was "published"...

Reply:
By that I mean before we merge the PR.

Reply:
I still think individual fields are better. In case we start adding more complexity later down the road (for example, the socket state), we can just add additional keys as we need to: this will give us room to grow.

With a single `status` field, we can't add any new statuses without breaking existing code which will rely on concrete semantics.

Reply:
:man_shrugging: Okay, as long as the provisional documentation is revised to stress that a non-`true` value for `connected` does NOT mean that it is possible or wise to attempt to call `reconnect()`. I would strongly encourage that all the values that `(QAbstractSocket::SocketState) ctelnet::getConnectionState()` can return are also provided in a `status` field... I guess if you don't want to do it here, it can always be added in a separate PR.